### ISSUE TITLE
observed consumer crash

### ISSUE THREAD

**Issue by bjustin-ibm** [openwhisk/openwhisk-package-kafka#35]
It appears there was either a network glitch or the kafka instance was unavailable during the poll. That being said, our code should handle this situation more gracefully.

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/KafkaFeedProvider/consumer.py", line 89, in run
    partition = self.consumer.poll(1000)
  File "/usr/local/lib/python2.7/site-packages/kafka/consumer/group.py", line 514, in poll
    records = self._poll_once(remaining)
  File "/usr/local/lib/python2.7/site-packages/kafka/consumer/group.py", line 541, in _poll_once
    self._coordinator.ensure_coordinator_known()
  File "/usr/local/lib/python2.7/site-packages/kafka/coordinator/base.py", line 221, in ensure_coordinator_known
    self._client.poll(future=metadata_update)
  File "/usr/local/lib/python2.7/site-packages/kafka/client_async.py", line 483, in poll
    self._maybe_connect(node_id)
  File "/usr/local/lib/python2.7/site-packages/kafka/client_async.py", line 322, in _maybe_connect
    conn.connect()
  File "/usr/local/lib/python2.7/site-packages/kafka/conn.py", line 304, in connect
    if self._try_handshake():
  File "/usr/local/lib/python2.7/site-packages/kafka/conn.py", line 366, in _try_handshake
    self._sock.do_handshake()
  File "/usr/local/lib/python2.7/ssl.py", line 830, in do_handshake
    self._sslobj.do_handshake()
error: [Errno 110] Operation timed out
```

**Comment by bjustin-ibm** [comment]
On the plus side, this crash was detected using the code in #27 👍

**Comment by bjustin-ibm** [comment]
Resolved by #54
